src/Yasumi/Provider/CommonHolidays.php

```diff
@@ -355,4 +355,98 @@ public function armisticeDay(
             $type
         );
     }
+
+    /**
+     * Calculates daylight saving time transitions.
+     *
+     * Daylight saving time is the practice of advancing clocks by one hour during summer months so evening daylight lasts even longer, while sacrificing normal sunrise times.
+     *
+     * The date of transition between standard time and daylight saving time differs from country to country and sometimes from year to year. Most countries outside Europe and North America do not observe daylight saving time.
+     *
+     * On the northern hemisphere, summer time starts around March/April. On the southern hemisphere it happens 6 months later.
+     *
+     * @param int    $year     the year for which Easter needs to be calculated
+     * @param string $timezone the timezone in which Easter is celebrated
+     * @param bool   $summer   whether to calculate the start of summer or winter time
+     */
+    protected function calculateSummerWinterTime($year, $timezone, $summer)
+    {
+        $zone = new DateTimeZone($timezone);
+
+        $transitions = $zone->getTransitions(mktime(0, 0, 0, 1, 1, $year), mktime(23, 59, 59, 12, 31, $year));
+
+        $transition = array_shift($transitions);
+        $dst = $transition['isdst'];
+
+        foreach ($transitions as $transition) {
```

Comment:
It seems you rely on the fact that the `getTransitions` function returns the transitions sorted by time, correct? If returned randomly, I think getting the transitions for summer and wintertime then might not work this way.

Reply:
AFAICT from the PHP source code, we can rely on the dates being returned in chronological order. It seems the underlying data structure is also sorted by date.

https://github.com/php/php-src/blob/ef255c9f0f2c2ad1ea224ed215edb00f5bc205bd/ext/date/php_date.c#L4013